### ISSUE TITLE
linter: tries to lint non-ts/js script blocks in Vue SFCs

### ISSUE THREAD

**Issue by ofekd** [oxc-project/oxc#10736]
### What version of Oxlint are you using?

0.16.0

### What command did you run?

oxlint . --fix -D correctness --ignore-path .gitignore

### What does your `.oxlintrc.json` config file look like?

No config file

### What happened?

Oxlint tries parsing the following block:

```
<script lang="gleam">
import gleam/io

pub fn hello () {
 io.debug("hello")
}
</script>
```

Which leads to unnecessary errors

**Comment by camc314** [comment]
fixed by #10740
